src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ActivatorUtilities.cs

```diff
@@ -73,7 +79,13 @@ public static class ActivatorUtilities
                 throw new InvalidOperationException(message);
             }
 
-            return bestMatcher.CreateInstance(provider);
+            if (largestLengthWithMultipleAcceptedCtors == bestLength)
+            {
+                string? message = $"multiple constructors for type '{instanceType}' were found with length {bestLength}.";
+                throw new InvalidOperationException(message);
+            }
+
+            return bestMatcher.CreateInstance(provider!);
```

Comment:
Why is this `!` valid? I think it would be better if we either threw an `ArgumentNullException` or actually made the parameter nullable.